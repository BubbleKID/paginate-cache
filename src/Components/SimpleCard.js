import React, { Component } from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import Card from "@material-ui/core/Card";
import CardActions from "@material-ui/core/CardActions";
import CardContent from "@material-ui/core/CardContent";
import Button from "@material-ui/core/Button";
import Typography from "@material-ui/core/Typography";

const styles = {
  card: {
    width: "300px",
    height: "240px"
  },
  title: {
    fontSize: "14px"
  },
  name: {
    marginTop: "15px"
  },
  pos: {
    marginBottom: "15px"
  },
  des: {
    height: "40px"
  }
};

class SimpleCard extends Component {
  render() {
    const { classes, card } = this.props;
    return (
      <Card className={classes.card}>
        <CardContent>
          <Typography
            className={classes.title}
            color="textSecondary"
            gutterBottom
            align="left"
          >
            {card.coreData.state}
          </Typography>
          <Typography
            className={classes.name}
            variant="h5"
            component="h2"
            align="left"
          >
            {card.coreData.number}
          </Typography>
          <Typography
            className={classes.pos}
            align="left"
            color="textSecondary"
          >
            Application: {card.coreData.application}
            <br />
            Assignee:{card.coreData.assignee}
          </Typography>
          <Typography component="p" align="left" className={classes.des}>
            {card.coreData.shortDescription.length > 66
              ? card.coreData.shortDescription.slice(0, 66).concat("......")
              : card.coreData.shortDescription}
          </Typography>
        </CardContent>
        <CardActions>
          <Button
            size="small"
            onClick={() => this.props.handleOpenDrawer(card)}
          >
            Learn More
          </Button>
        </CardActions>
      </Card>
    );
  }
}

SimpleCard.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(styles)(SimpleCard);
