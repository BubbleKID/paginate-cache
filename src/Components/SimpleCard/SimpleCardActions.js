export const setCurrentCard= currCardData => ({
  type: "SET_CURRENTCARDS",
  currCardData
});
