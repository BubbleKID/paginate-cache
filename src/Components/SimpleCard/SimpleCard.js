import React, { Component } from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import Card from "@material-ui/core/Card";
import CardActions from "@material-ui/core/CardActions";
import CardContent from "@material-ui/core/CardContent";
import Button from "@material-ui/core/Button";
import Typography from "@material-ui/core/Typography";
import { withRouter } from "react-router-dom";
import * as actions from "../SimpleCard/SimpleCardActions";
import { connect } from "react-redux";

const styles = {
  card: {
    width: "300px",
    height: "240px"
  },
  title: {
    fontSize: "14px"
  },
  name: {
    marginTop: "15px"
  },
  pos: {
    marginBottom: "15px"
  },
  des: {
    height: "40px"
  }
};

class SimpleCard extends Component {
  render() {
    const { classes, card } = this.props;
    return (
      <Card className={classes.card}>
        <CardContent>
          <Typography
            className={classes.title}
            color="textSecondary"
            gutterBottom
            align="left"
          >
            {card.coreData.state}
          </Typography>
          <Typography
            className={classes.name}
            variant="h5"
            component="h2"
            align="left"
          >
            {card.coreData.number}
          </Typography>
          <Typography
            className={classes.pos}
            align="left"
            color="textSecondary"
          >
            Application: {card.coreData.application}
            <br />
            Assignee:{card.coreData.assignee}
          </Typography>
          <Typography component="p" align="left" className={classes.des}>
            {card.coreData.shortDescription.length > 66
              ? card.coreData.shortDescription.slice(0, 66).concat("......")
              : card.coreData.shortDescription}
          </Typography>
        </CardContent>
        <CardActions>
          <Button
            size="small"
            onClick={() => {
              this.props.dispatch(actions.setCurrentCard(card));
              this.props.history.push("/card/" + card.coreData.number);
            }}
          >
            Learn More
          </Button>
        </CardActions>
      </Card>
    );
  }
}

SimpleCard.propTypes = {
  classes: PropTypes.object.isRequired
};

const mapStateToProps = state => ({
  currCard: state.currCard
});

export default connect(mapStateToProps)(
  withRouter(withStyles(styles)(SimpleCard))
);
