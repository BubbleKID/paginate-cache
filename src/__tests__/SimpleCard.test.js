import React from "react";
import { shallow } from "enzyme";
import SimpleCard from "../Components/SimpleCard";
describe("SimpleCard", () => {
  it("should render correctly", () => {
    const component = shallow(<SimpleCard />);
    expect(component).toMatchSnapshot();
  });
});
