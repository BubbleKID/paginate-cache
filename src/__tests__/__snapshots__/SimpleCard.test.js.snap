// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleCard should render correctly 1`] = `
<SimpleCard
  classes={
    Object {
      "card": "SimpleCard-card-1",
      "des": "SimpleCard-des-5",
      "name": "SimpleCard-name-3",
      "pos": "SimpleCard-pos-4",
      "title": "SimpleCard-title-2",
    }
  }
/>
`;
